### PR TITLE
[MIG] bi_sql_editor v10

### PR DESCRIPTION
**Pull request by serpentcs-dev1**
Migrates **bi_sql_editor** from [v9](https://github.com/OCA/reporting-engine/tree/9.0/bi_sql_editor) to v10

### CODE REVIEW

**Comment by serpentcs-dev1** [comment]
No significant change needed. Works fine!!

@legalsylvain @jbeficent Could you please take a look?

**Comment by legalsylvain** [comment]
Hi, you should set a custom dependencies to the current PR against server-tools in your oca dependencies file. Otherwise, it will not be possible to test on runbot.

**Comment by JordiBForgeFlow** [comment]
@darshan-serpent See https://github.com/OCA/maintainer-quality-tools/blob/master/sample_files/oca_dependencies.txt#L15

**Review by AaronHForgeFlow — COMMENTED** [review]
Functional tests. Basic flows work. Maybe you can include this improvement: https://github.com/OCA/reporting-engine/pull/156

**Comment by AaronHForgeFlow** [comment]
@legalsylvain can you please review?

**Review by AaronHForgeFlow — COMMENTED** [review]
Can't uninstall the module:
image

The uninstall hook is not working properly. I'll keep reviewing

File: bi_sql_editor/tests/test_bi_sql_view.py
Comment (AaronHForgeFlow) [inline bi_sql_editor/tests/test_bi_sql_view.py:40]: Test are good. Just add this here to make it not fail:
`'notify_email': 'none',`

Running the test multiple times won't create multiple bi_sql_view records, that's good. :smile:

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]

**Comment by AaronHForgeFlow** [comment]
@darshan-serpent this is merged! https://github.com/OCA/server-tools/pull/961 :tada: you can remove the explicit dependency now: https://github.com/OCA/reporting-engine/pull/162/commits/a2e653346b81f63f91fa058d48f4d6e58060f1de

**Comment by LoisRForgeFlow** [comment]
Any update here? Is this PR ready for review??

**Comment by serpentcs-dev1** [comment]
@lreficent Yes, more reviews needed!

**Review by LoisRForgeFlow — APPROVED** [review]
Functional review seems good! :+1:

File: bi_sql_editor/views/view_bi_sql_view.xml
Comment (legalsylvain) [inline bi_sql_editor/views/view_bi_sql_view.xml:32]: Could you remove the oe_highlight class, as it is not a required step. (just an extra possible action).

File: bi_sql_editor/models/bi_sql_view.py
Comment (legalsylvain) [inline bi_sql_editor/models/bi_sql_view.py:578]: Hi.

good inovation that can be interesting in some cases. I think that in a next step, we could add better a limit to _execute_sql_request() instead of limit the parse of the result. (because if there are 2 billion of result, this button will be very slow).

I'm not very fan of the display (by a UserError), but I think that making a great display is a big work (maybe using this module https://github.com/OCA/web/tree/8.0/web_widget_x2many_2d_matrix)
So let's go approving this PR and improving when we'll have time !
image

**Review by legalsylvain — CHANGES_REQUESTED** [review]
- a minor remark about oe_highlight class.
- https://github.com/OCA/reporting-engine/pull/162#pullrequestreview-66767226 remarks to fix.

Otherwise, LGTM. 

Change Code Review + Runbot Test.

File: bi_sql_editor/views/view_bi_sql_view.xml
Comment (astirpe) [inline bi_sql_editor/views/view_bi_sql_view.xml:13]: Please replace the `colors=` syntax with the new `decoration-*` way to define colors of the rows. I think the `colors=` way is not even working anymore.

File: bi_sql_editor/views/view_bi_sql_view.xml
Comment (astirpe) [inline bi_sql_editor/views/view_bi_sql_view.xml:73]: Please replace the `colors=` syntax with the new `decoration-*` syntax.

File: bi_sql_editor/README.rst
Comment (astirpe) [inline bi_sql_editor/README.rst:2]: Use the .png image and the https for the links. See the updated template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: bi_sql_editor/__manifest__.py
Comment (astirpe) [inline bi_sql_editor/__manifest__.py:8]: Use single quotes (instead of double quotes) to be consistent with all the rest.

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by CasVissers-360ERP** [comment]
Hi, 

I get a traceback when running the cron:
```

Traceback (most recent call last):
  File "/opt/odoo/odoo/odoo/odoo/addons/base/ir/ir_cron.py", line 126, in _callback
    getattr(model, method_name)(*args)
TypeError: button_refresh_materialized_view() takes exactly 1 argument (2 given)
``

**Comment by espo-tony** [comment]
Hi, @CasVissers 
The cron is called passing a `bi.sql.view` record list as argument, which is different than invoking the method on a `bi.sql.view` recordset (as requested from the method `button_refresh_materialized_view` signature, which takes no arguments and has decorated with `api.multi` ).

Considering the way this method is used, I suggest to change it into a model method (`api.model` decorated method) which takes as parameter a `bi.sql.view` recordset.

**Comment by CasVissers-360ERP** [comment]
Hi,

When running on update all on a database with the module installed:
ProgrammingError: "" is not a table or foreign table

Which blocks the process untill the model is removed.

**Comment by pedrobaeza** [comment]
Superseeded by #206

**Pull request closed without merging**
